src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuardsNew.java

```diff
@@ -0,0 +1,977 @@
+    {
+        this.followPlayer = player;
+    }
+
+    /**
+     * If no vision multiplier give health bonus.
+     *
+     * @return the bonus health.
+     */
+    private int getBonusHealth()
+    {
+        if (getBuildingLevel() > MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return (getBuildingLevel() - MAX_VISION_BONUS_MULTIPLIER) * HEALTH_MULTIPLIER;
+        }
+        return 0;
+    }
+
+    /**
+     * Adds new patrolTargets.
+     *
+     * @param target the target to add
+     */
+    public void addPatrolTargets(final BlockPos target)
+    {
+        this.patrolTargets.add(target);
+        this.markDirty();
+    }
+
+    /**
+     * Resets the patrolTargets list.
+     */
+    public void resetPatrolTargets()
+    {
+        this.patrolTargets = new ArrayList<>();
+        this.markDirty();
+    }
+
+    /**
+     * Getter for the bonus vision.
+     *
+     * @return an integer for the additional range.
+     */
+    public int getBonusVision()
+    {
+        if (getBuildingLevel() <= MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return getBuildingLevel() * VISION_BONUS;
+        }
+        return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
+    }
+
+    /**
+     * Populates the mobs list from the ForgeRegistries.
+     *
+     * @return the list of MobEntrys to attack.
+     */
+    public List<MobEntryView> calculateMobs()
+    {
+        final List<MobEntryView> mobs = new ArrayList<>();
+
+        int i = 0;
+        for (final EntityEntry entry : ForgeRegistries.ENTITIES.getValuesCollection())
+        {
+            if (EntityMob.class.isAssignableFrom(entry.getEntityClass()))
+            {
+                i++;
+                mobs.add(new MobEntryView(entry.getRegistryName(), true, i));
+            }
+            else
+            {
+                for (String location : Configurations.gameplay.guardResourceLocations)
+                {
+                    if (entry.getRegistryName() != null && entry.getRegistryName().toString().equals(location))
+                    {
+                        i++;
+                        mobs.add(new MobEntryView(entry.getRegistryName(), true, i));
+                    }
+                }
+            }
+        }
+
+        getColony().getPackageManager().getSubscribers().forEach(player -> MineColonies
+                                                                             .getNetwork()
+                                                                             .sendTo(new GuardMobAttackListMessage(getColony().getID(),
+                                                                               getID(),
+                                                                               mobsToAttack),
+                                                                               player));
+
+        return mobs;
+    }
+
+    /**
+     * The client view for the Guard building.
+     */
+    public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 127 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)